.github/workflows/my_ci_pipeline.yaml

```diff
@@ -8,9 +8,28 @@ on:
       - 'README.md'
 jobs:
   build:
-    name: Build and test
+    name: Build and Test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - run: docker build --target test --tag todo_app:test -f dockerfiles/Dockerfile.todo_app .
-    - run: docker run todo_app:test
\ No newline at end of file
+    - run: docker run todo_app:test
+
+  deploy: 
+    name: Docker Hub Push & Triggers App Service Deploy
+    runs-on: ubuntu-latest
+    needs: build 
+    if: ${{github.event_name=='push' && github.ref=='refs/heads/ASAMA_0009'}}
```

Comment:
Like you mentioned you'll want to change the branch name back to `main` after merging the PR

Reply:
Thanks as I was a bit confused inititally looking at it